spec/timezone.html

```diff
@@ -626,9 +626,17 @@ <h1>BuiltinTimeZoneGetInstantFor ( _timeZone_, _dateTime_, _disambiguation_ )</h
       <h1>GetPossibleInstantsFor ( _timeZone_, _dateTime_ )</h1>
       <emu-alg>
         1. Let _possibleInstants_ be ? Invoke(_timeZone_, *"getPossibleInstantsFor"*, « _dateTime_ »).
-        1. Let _list_ be ? CreateListFromArrayLike(_possibleInstants_, « Object »).
-        1. For each element _instant_ in _list_ in List order, do
-          1. Perform ? RequireInternalSlot(_instant_, [[InitializedTemporalInstant]]).
+        1. Let _iteratorRecord_ be ? GetIterator(_possibleInstants_, ~sync~).
+        1. Let _list_ be a new empty List.
+        1. Let _next_ be *true*.
+        1. Repeat, while _next_ is not *false*,
+          1. Set _next_ to ? IteratorStep(_iteratorRecord_).
+          1. If _next_ is not *false*, then
+            1. Let _nextValue_ be ? IteratorValue(_next_).
+            1. If Type(_nextValue_) is not Object or _nextValue_ does not have an [[InitializedTemporalInstant]] internal slot, then
+              1. Let _completion_ be ThrowCompletion(a newly created *TypeError* object).
+              1. Return ? IteratorClose(_iteratorRecord_, _completion_).
```

Comment:
Do IteratorStep and IteratorValue's abrupt return need to also call IteratorClose?

Reply:
No; if those throw it represents error an in the _iterator itself_, rather than the iterator working correctly but the consumer having an issue.

Reply:
thanks, i wasn't sure about that (and didn't check the main spec first)

Reply:
They don't do so in https://tc39.es/ecma262/#sec-iterabletolist, so is this something that should be resolved at the ecma262 level?

Reply:
Oh, I guess my comment crossed with Kevin's; resolving.